content/post/2022-07-28-gsoc-ldr426-intro/index.md

```diff
@@ -0,0 +1,54 @@
+
+![The IOOS Logo - The U.S. Integrated Ocean Observing System (IOOS)](https://ioos.us/images/IOOS_Emblem_Tertiary_B_RGB.png)
+
+One of the main focuses of the [Echospace](https://uw-echospace.github.io/author/echospace/) team is sampling and interpretation of ocean acoustic data. [Echopype](https://github.com/OSOceanAcoustics/echopype) sits in the middle, extracts raw data from the cloud or file server, converts them to netCDF or Zarr, and performs denoising and calibration. Another job is to interpret, where I give my effort to build a library called echoshader that can help oceanographers discover certain patterns from it. [Echoshader](https://github.com/OSOceanAcoustics/echoshader), an open source project, aims to enhance the ability to interactively visualize large amounts of cloud-based data to accelerate the data exploration and discovery process. Ocean sonar data are generated from echopype, which handles the normalization, preprocessing and organization of echo data. Echoshader will be developed in parallel with the ongoing development of echopype. 
+
+As a participant of GSoC, I am developing the main APIs of echoshader based on the [HoloViz](https://holoviz.org/) suite of tools, test configuration for using echoshader widgets in Panel dashboards, and create Jupyter notebooks to demo use of the combination of tools. 
+
+![Deploying Panel (Holoviz) dashboards using Heroku Container Registry | by  Ali Shahid | Towards Data Science](https://miro.medium.com/max/1400/1*xQEm58a7c_g1Go9G5NMyuw.jpeg)
+
+Before starting coding, I read lots of documents to find the most suitable tool. Although there are many excellent and fantastic visualizing libraries with Python, such as plotly and bokh, they can not process xarray directly, which is a kind of multidimensional labeled data massively used in echopype.  Then I locked my eyes on HoloViz ecosystem, whose tools and examples generally work with any Python standard data types (lists, dictionaries, etc.), plus Pandas or Dask DataFrames and NumPy, Xarray, or Dask arrays.
+After determining which type of tool to use, I began to read a user guide about HoloViz libraries. There are several libraries mainly used in echoshader: hvplot, Holoviews, GeoViews and Panel.  Hvplot and HoloViews declare objects for instantly visualizable data, building Bokeh plots from convenient high-level specifications. GeoViews visualize geographic data corresponding to ship survey datasets. Panel assembles grams and control widgets from these different libraries into a layout which could be displayed in a Jupyter notebook and in a standalone servable dashboard.  In addition to HoloViz libraries, PyVista and other libraries are involved for 3D extension, which also fit well in panel layout. Also, benchmarking and doc work are required for each module.
+
```

Comment:
Below are some screenshots of the different visualization functionalities I am developing:

![2d_echogram](https://user-images.githubusercontent.com/15334215/186999651-76081a29-11f8-4d37-b3a9-fca0ad49a03c.png)

![tracks](https://user-images.githubusercontent.com/15334215/186999662-ba744a49-b02e-4451-a716-f8c8df654053.png)

![curtain](https://user-images.githubusercontent.com/15334215/186999678-2bf77985-aab3-42f8-88f9-1f2c78d3b2eb.png)